lib/inc/internal/token.hpp

```diff
@@ -39,4 +39,7 @@ namespace hocon {
         std::string _debug_string;
     };
 
+    using token_list = std::vector<std::shared_ptr<token>>;
+    using shared_token = std::shared_ptr<token>;
```

Comment:
I'd make this `shared_ptr<const token>`, and declare `token_list` in terms of `vector<shared_token>`.